### PR TITLE
Implement coalesce functions.

### PR DESCRIPTION
**Pull request by jmalloc**
I've adopted a convention of naming whereby functions that target `time.Time` values are suffixed with `T`, whereas those that are for `time.Duration` values do not have any suffix.

### CODE REVIEW

**Comment by jmalloc** [comment]
I think I will go back and change these tests to examples.

**Comment by jmalloc** [comment]
Actually, no... there's no nice way to "expect" panics using examples, which I need to test for. I'll leave this as is, but add some examples where appropriate in the future.

File: coalesce.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package linger
+
+import "time"
+
+// Coalesce returns the first of its arguments that is positive.
+//
+// If none of the arguments are positive, v is th zero-value and ok is false.
```
Comment (koden-km) [inline coalesce.go]:
```suggestion
// If none of the arguments are positive, v is the zero-value and ok is false.
```

File: coalesce.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// MustCoalesce returns the first of its arguments that is positive.
+//
+// It panics if none of the arguments are positive.
+func MustCoalesce(values ...time.Duration) time.Duration {
+	return MustFirst(Positive, values...)
+}
+
+// CoalesceT returns the first of its arguments that is non-zero.
+//
+// If none of the arguments are non-zero, v is th zero-value and ok is false.
```
Comment (koden-km) [inline coalesce.go]:
```suggestion
// If none of the arguments are non-zero, v is the zero-value and ok is false.
```

**Review by koden-km — APPROVED** [review]
Just a couple of doc block typos.

**Pull request merged**